spec/taxjar/client_spec.rb

```diff
@@ -13,6 +13,34 @@
     end
   end
 
+  describe "#set_api_config" do
+    it 'sets new api key' do
+      client = Taxjar::Client.new(api_key: 'AK')
+      client.set_api_config('api_key', 'ZZ')
+      expect(client.api_key).to eq('ZZ')
+    end
+
+    it 'sets new api url' do
+      client = Taxjar::Client.new(api_key: 'AK')
+      client.set_api_config('api_url', 'https://api.sandbox.taxjar.com')
+      expect(client.api_url).to eq('https://api.sandbox.taxjar.com')
+    end
+    
+    it 'sets new custom headers' do
+      client = Taxjar::Client.new(api_key: 'AK')
+      client.set_api_config('headers', { 'X-TJ-Expected-Response' => 422 })
+      expect(client.headers).to eq({ 'X-TJ-Expected-Response' => 422 })
```

Comment:
this means auth header is lost if I want to add custom header?

Reply:
@slamorsi Auth and user agent headers will always be included in the `set_request_headers` method, `client.headers` just refers to your custom headers